### DISCUSSION TITLE
Switch language (keyboard) problem

### DISCUSSION THREAD

**Discussion by vstath** [regolith-linux/regolith-desktop#965] [Q&A] [answered]
Hello,

I have a new installation on Ubuntu 22.04 and something is going wrong. I cannot switch keyboards (input sources). The hotkeys don't work. Neither defaults that appear in the Keyboard settings page, not the Regolith ones I used to have (mod+alt+backspace IIRC). 

Note that when I reorder the input sources in the settings page, all work fine :-) Of course, that's not a solution and it's ultra annoying. 

I have resorted to go to a terminal and change keyboard layout manually: 

```
gsettings set org.gnome.desktop.input-sources sources "[('xkb', 'de'), ('xkb','us')]"
```
and then back to US

Additionally, I don't see any flags or DE/EN or similar in the bottom bar. Not sure if it should be there or not. 

Any ideas what I should look into? It's very frustrating, as you can imagine :-D

**Comment by vstath** [comment]
Should regolith show language on the bottom bar? Is this something that's included? What's the base behavior regarding languages? Or we don't really care as the focus is on EN speakers and the rest can figure it out on their own (if so)? I would appreciate an answer so I can focus my energy and attention properly. 

PS I really like Regolith but it's a pity it hasn't reached mainstream. Maybe the team could offer paid consulting / support / whatever services to fund the project?

**Comment by kgilmer** [comment, marked as answer]
Hi @vstath ,

This package should help with showing the active keyboard layout in the bar (X11 only, see i3rstatus-rs for sway): i3xrocks-keyboard-layout

Regarding a better user experience with multi-language layouts, yes it could definitely be improved.  I don't use multiple layouts or languages so I don't have much understanding of what is needed her.  Given that you have found command line programs to configure the layout, you might consider adding them as keybindings in your i3 config.

**Reply by vstath** [reply]
Hi @kgilmer, 

unfortunately I had tried that in the past and was unable to make it work because the keyboard I switch to is not recognized as a mapping (same if I do **$mod+[** for example - it says "unknown key or button xyz"). 

After your recommendation I looked at the existing mappings and I decided to replace the grave/shift-grave combination. It's a bit of a hack but it now works fine. Thank you. 

Regarding multiple languages, the idea is simple. You go to settings -> keyboard and you add another keyboard (for example Greek, to make it non Latin so it's obvious it switches back and forth). You have the hotkey combination, that allows you to switch languages in the keyboard. That's all. 

I have read a similar response you gave couple years back; I'm happy to assist in this (by testing / trying things you tell me) if you would like to. In my original Regolith 1.x, the combination of CMD + Alt + backspace was the default IIRC. Currently, no combination works for me no matter what I try to set. It used to work fine on the stock ubuntu 22.04 before installing regolith linux on top of it.

Regarding the sway I am not sure how to do it but I will poke around and try to find where to configuration is and what I need to do. I assume this is default i3 behavior so I will search on their docs. Very much appreciated your responses.

**Reply by kgilmer** [reply]
Here are the docs for they keyboard-layout block for sway: https://github.com/greshake/i3status-rust/blob/master/src/blocks/keyboard_layout.rs
